### PR TITLE
Feature 2 swimmer search by location

### PR DESCRIPTION
**Pull request by katiefugate**
Screen Shot 2021-05-27 at 3 35 36 PM
Screen Shot 2021-05-27 at 3 34 34 PM
Screen Shot 2021-05-27 at 3 34 49 PM
Screen Shot 2021-05-27 at 3 44 12 PM
Screen Shot 2021-05-27 at 3 47 58 PM

### CODE REVIEW

File: client/pages/home.jsx [hunk 1]
```diff
@@ -1,8 +1,32 @@
...
-  );
+class Home extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(event) {
+
+  }
```
Comment (thebearingedge) [inline client/pages/home.jsx:11]: You can lose these methods.

File: client/pages/search-page.jsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import parseRoute from '../lib/parse-route';
+
+class Search extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.state = {
+      route: parseRoute(window.location.hash)
+    };
```
Comment (thebearingedge) [inline client/pages/search-page.jsx:10]: I don't think this state is necessary, but you should make sure that you are keeping track of the user's input value in state.

File: client/app.jsx [hunk 3]
```diff
@@ -2,15 +2,65 @@ import React from 'react';
...
+      < Search />
+      < Navbar />
+      </>
+      );
+    }
+    if (path === 'search-results') {
+      return (
+        <>
+        < Header />
+        < SearchResults />
```
Comment (thebearingedge) [inline client/app.jsx:52]: You should pass in your app's route param here for the search term.

File: client/pages/search-results.jsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    this.state = {
+      route: parseRoute(window.location.hash),
+      poolList: null,
+      isLoading: true
+    };
+  }
+
+  componentDidMount() {
+    const { params } = this.state.route;
+    const location = params.get('location');
```
Comment (thebearingedge) [inline client/pages/search-results.jsx:17]: Try to leave all of the route parsing in App and pass dynamic values via props.

File: server/index.js [hunk 5]
```diff
@@ -59,6 +59,26 @@ app.post('/api/hosts', uploadsMiddleware, (req, res, next) => {
...
+           "image",
+           "poolId"
+      from "pools"
+    where "location" = $1`;
+  const params = [location];
+  db.query(sql, params)
+    .then(result => {
+      if (!result.rows[0]) {
+        throw new ClientError(400, `No pools found in ${location}`);
+      }
```
Comment (thebearingedge) [inline server/index.js:76]: An empty list is fine. The request was successful and (maybe) later, there could be results for that location. 4xx errors mean that the client will _never_ get what they want with that request. Just show a "no results found" screen if the list is empty.

**Review by thebearingedge — CHANGES_REQUESTED** [review]
Ok, great work so far, just a bit of clean up

**Comment by katiefugate** [comment]
Screen Shot 2021-05-27 at 4 43 16 PM
Screen Shot 2021-05-27 at 4 43 33 PM

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**